src/components/Sidebar.tsx

```diff
@@ -0,0 +1,37 @@
+import { Bars3Icon, XMarkIcon } from "@heroicons/react/24/solid";
+import { useState } from "react";
+import Settings from "./Settings";
+
+const Sidebar = () => {
+  const [showSidebar, setShowSidebar] = useState(false);
+
+  return (
+    <>
+      {showSidebar ? (
+        <button
+          className="flex text-3xl text-white items-center cursor-pointer fixed right-10 top-6 z-50 right-5"
+          onClick={() => setShowSidebar(!showSidebar)}
+        >
+          <XMarkIcon className="w-12 fill-warning-dark flex-none stroke-white" />
+        </button>
+      ) : (
+        <Bars3Icon className="w-12 fill-warning-dark flex-none fill-white" onClick={() => setShowSidebar(!showSidebar)} />
+      )}
+
+        
+
+
+      <div
+        className={`top-0 right-0 w-[100vw] md:w-[35vw] bg-[#D41B2C] p-10 pl-20 text-white fixed h-full z-40  ease-in-out duration-300 ${showSidebar ? "translate-x-0 " : "translate-x-full"
```

Comment:
I'd avoid hardcoding styles as much as possible, so here I would replace w-[100vw] with w-screen, and replace the bg with one of our predefined colors in our tailwind.config.json file